addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -57,7 +57,10 @@ function factory(dependencies) {
          * Send the attachment for the browser to download.
          */
         download() {
-            this.env.services.navigate(`/web/content/ir.attachment/${this.id}/datas`, { download: true });
+            let downloadLink = document.createElement('a');
+            downloadLink.setAttribute('href', `/web/content/ir.attachment/${this.id}/datas`);
```

Comment:
This url is wrong for guest, but out of scope, it will be fixed in task-2664842.